### ISSUE TITLE
Reset button should cause an immediate reset, not wait for the agent to finish the current step

### ISSUE THREAD

**Issue by PierrunoYT** [All-Hands-AI/OpenHands#2305]
**What problem or use case are you trying to solve?
I would like to stop the agent immediately and cancel the current task.

**Describe the UX of the solution you want**
There should be a stop button next to the pause button

**Have ideas for the technical implementation?

**Describe alternatives you have considered**

**Additional context**

**Comment by tobitege** [comment]
Absolutely +1 on this!

**Comment by iFurySt** [comment]
is the reset button cover this situation?

**Comment by tobitege** [comment]
@iFurySt I'm not sure what "cover" means here?

**Comment by iFurySt** [comment]
the reset button we currently have stops the agent and starts a new session. i think this already covers the stop scenario. wdyt?

**Comment by tobitege** [comment]
The problem right now is, that it is not immediate. Not sure it should reset the session, it should just any operation the agent is currently doing, including *whilst* waiting on an LLM reponse.

**Comment by iFurySt** [comment]
that's correct, we are unable to stop the agent immediately after clicking the button.

**Comment by PierrunoYT** [comment]
Is this active?

**Comment by enyst** [comment]
Yes, thanks for keeping it active.

**Comment by shubhamofbce** [comment]
Hi @tobitege,

I am planning to work on this, but I noticed that the reset (start a new task) button is working properly and immediately resets everything. However, the pause button is not pausing the action immediately; instead, it waits for the current task to finish before pausing.

So Is this issue about the pause button or reset(start a new task button)? Let me know if I am understanding something incorrect.
cc: @neubig

**Comment by tobitege** [comment]
Hey @shubhamofbce 
it's about a hard-stop button, basically, that would terminate a running operation (completion, terminal, browser) and give back control to the user in the UI (doesn't have to be all variants at once).

A while back, when we still had the DockerSSHbox, I had something for this cooked up to a certain point, but then a lot of stuff got changed around and I closed it: #2602 
Maybe you can take some ideas/sources from it (like the UI parts?). I think I even had some terminal chunked-streaming in (CancellableStream, in the old plugin mixin class I think) and other bits and pieces.

Currently I'd recommend to first look into the communication part of the EventStreamRuntime.
It right now sort of by design only allows a single command, non-async, so as long as some operation is running, it won't react to any other UI element (you'd need to check that). So here we may have to look into making it a) fully async and b) maintain a queue of requests.

Feel free to plan out what parts you'd like to look into/implement and we'll certainly assist you if needed, thank you! 🙏

**Comment by neubig** [comment]
@amanape just clarifying, was this fixed by https://github.com/All-Hands-AI/OpenHands/pull/3604 ?

**Comment by mamoodi** [comment]
Is someone able to comment on this? There is another issue that asks for the reset and pause buttons to be more responsive. Will that one take care of this one as well?

**Comment by mamoodi** [comment]
This has been fixed in latest 0.15.0 release.
